src/OpenLoco/src/Drawing/SoftwareDrawingContext.cpp

```diff
@@ -1509,16 +1510,129 @@ namespace OpenLoco::Drawing
         }
 
         // 0x004474BA
+        // 0x00494E33
+        // ax: left
+        // bx: right
+        // bp: width
+        // cx: top
+        // dx: bottom
+        // ebp: colour | enumValue(flags)
+        // edi: rt
         static void drawRectImpl(Gfx::RenderTarget& rt, int16_t left, int16_t top, int16_t right, int16_t bottom, uint8_t colour, RectFlags flags)
         {
-            registers regs;
-            regs.ax = left;
-            regs.bx = right;
-            regs.cx = top;
-            regs.dx = bottom;
-            regs.ebp = colour | enumValue(flags);
-            regs.edi = X86Pointer(&rt);
-            call(0x004474BA, regs);
+            if (left > right)
+            {
+                return;
+            }
+            if (top > bottom)
+            {
+                return;
+            }
+
+            Rect startRect = Rect::fromLTRB(left, top, right, bottom);
+            Rect renderTargetRect = Rect(rt.x, rt.y, rt.width, rt.height);
+
+            if (!startRect.intersects(renderTargetRect))
+            {
+                return;
+            }
+
+            uint16_t crossPattern = 0;
+
+            int32_t leftX = left - rt.x;
```

Comment:
Are you able to use startRect.intersect for this instead? (And just copy paste the bits that are needed for the crossPattern into the crossPattern branch)